### PR TITLE
Various fixes from #4117

### PR DESCRIPTION
**Pull request by ondrejholy**
Proposing various fixes as a side-effect of debugging https://github.com/FreeRDP/FreeRDP/issues/4117

### CODE REVIEW

File: client/X11/xf_graphics.c [hunk 1]
```diff
@@ -202,10 +202,16 @@ static BOOL xf_Bitmap_SetSurface(rdpContext* context, rdpBitmap* bitmap,
...
 	xfContext* xfc = (xfContext*) context;
+
+	if (!context)
+		return FALSE;
+
 	xf_lock_x11(xfc, FALSE);
 
 	if (primary)
 		xfc->drawing = xfc->primary;
+	else if (!bitmap)
```
Comment (hardening) [inline client/X11/xf_graphics.c]: this check must be done some line above, or you leave the function with X11 locked. 
I'd be curious to see when that situation happens (I mean perhaps the check is to be done by the callers ?)

File: libfreerdp/core/orders.c [hunk 2]
```diff
@@ -2460,7 +2460,7 @@ static BOOL update_read_create_offscreen_bitmap_order(wStream* s,
 		if (deleteList->cIndices > deleteList->sIndices)
 		{
 			UINT16* new_indices;
-			new_indices = (UINT16*)realloc(deleteList->indices, deleteList->sIndices * 2);
+			new_indices = (UINT16*)realloc(deleteList->indices, deleteList->cIndices * 2);
```
Comment (hardening) [inline libfreerdp/core/orders.c]: good catch !

File: client/X11/xf_graphics.c (hunk 1 above)
Comment (ondrejholy) [reply client/X11/xf_graphics.c]: Dammit, sorry, updated. Will be also checked by callers thanks to the following patch...

**Comment by ondrejholy** [comment]
@hardening Is anything expected from my side?

**Pull request merged**